app/models/merchant.rb

```diff
@@ -6,6 +6,20 @@ def self.random
     order("RANDOM()").first
   end
 
+  def revenue(date)
+    invoices_by_date(date).joins(:transactions, :invoice_items)
+    .merge(Transaction.success)
+    .sum("invoice_items.unit_price * invoice_items.quantity")
+  end
+
+  def invoices_by_date(date)
```

Comment:
the only other way i can think of to accomplish this would be to call different methods in the controller - so if `params[:date]` then call maybe `revenue_by_date` or else just call `revenue`? 

i like what you have here though. this definitely accomplishes your goal 👍